admin/server/service.go

```diff
@@ -0,0 +1,280 @@
+package server
+
+import (
+	"context"
+
+	"github.com/rilldata/rill/admin/database"
+	"github.com/rilldata/rill/admin/server/auth"
+	adminv1 "github.com/rilldata/rill/proto/gen/rill/admin/v1"
+	"github.com/rilldata/rill/runtime/pkg/observability"
+	"go.opentelemetry.io/otel/attribute"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// CreateService creates a new service account.
+func (s *Server) CreateService(ctx context.Context, req *adminv1.CreateServiceRequest) (*adminv1.CreateServiceResponse, error) {
+	observability.AddRequestAttributes(ctx,
+		attribute.String("args.name", req.Name),
+		attribute.String("args.OrganizationName", req.OrganizationName),
+	)
+
+	// Check the request is made by an authenticated user
+	claims := auth.GetClaims(ctx)
+	if claims.OwnerType() != auth.OwnerTypeUser {
+		return nil, status.Error(codes.Unauthenticated, "not authenticated as a user")
+	}
```

Comment:
No need to check the call is made by a user – that's only needed when `claims.OwnerID()` is assumed to be a user ID